src/Adapter/Cart/CartPresenter.php

```diff
@@ -215,71 +215,92 @@ public function present($cart)
 
         $total_excluding_tax = $cart->getOrderTotal(false);
         $total_including_tax = $cart->getOrderTotal(true);
-        $total_discount      = $cart->getOrderTotal(true, Cart::ONLY_DISCOUNTS);
+        $total_discount = $cart->getOrderTotal(true, Cart::ONLY_DISCOUNTS);
 
-        if (Configuration::get('PS_TAX_DISPLAY')) {
-            $subtotals['tax'] = [
-                'type' => 'tax',
-                'label' => $this->translator->trans('Tax', [], 'Cart'),
-                'amount' => $this->priceFormatter->format(
-                    $total_including_tax - $total_excluding_tax
-                ),
-            ];
-        }
+        $subtotals['products'] = [
+            'type' => 'products',
+            'label' => $this->translator->trans('Products', [], 'Cart'),
+            'amount' => $cart->getOrderTotal(true, Cart::ONLY_PRODUCTS),
+            'value' =>  $this->priceFormatter->format(($cart->getOrderTotal(true, Cart::ONLY_PRODUCTS)))
+        ];
 
         if ($cart->gift) {
+            $giftWrappingPrice = ($cart->getGiftWrappingPrice($this->includeTaxes()) != 0)
+                ? $cart->getGiftWrappingPrice($this->includeTaxes())
+                : 0;
+
             $subtotals['gift_wrapping'] = [
                 'type' => 'gift_wrapping',
                 'label' => $this->translator->trans('Gift wrapping', [], 'Cart'),
-                'amount' => $cart->getGiftWrappingPrice($this->includeTaxes()) != 0
-                                ? $this->priceFormatter->format($cart->getGiftWrappingPrice($this->includeTaxes()))
-                                : $this->translator->trans('Free', [], 'Cart'),
+                'amount' => $giftWrappingPrice,
+                'value' => ($giftWrappingPrice > 0)
+                    ? $this->priceFormatter->format($giftWrappingPrice)
+                    : $this->translator->trans('Free', [], 'Cart'),
             ];
         }
 
+        $subtotals['discounts'] = [
+            'type' => 'discount',
+            'label' => $this->translator->trans('Discount', [], 'Cart'),
+            'amount' => $total_discount,
+            'value' => $this->priceFormatter->format($total_discount),
+        ];
+
         $shipping_cost = $cart->getTotalShippingCost(null, $this->includeTaxes());
         $subtotals['shipping'] = [
             'type' => 'shipping',
             'label' => $this->translator->trans('Shipping', [], 'Cart'),
-            'amount' => $shipping_cost != 0
+            'amount' => $shipping_cost,
+            'value' => $shipping_cost != 0
                             ? $this->priceFormatter->format($shipping_cost)
                             : $this->translator->trans('Free', [], 'Cart'),
```

Comment:
also, `[]` should be `array()` you can use phcsfixer with `long_array_syntax` enabled :)